projects/igniteui-angular/src/lib/grids/grid-base.directive.ts

```diff
@@ -3316,23 +3316,27 @@ export abstract class IgxGridBaseDirective extends DisplayDensityBase implements
     }
 
     constructor(
-        public validation: IgxGridValidationService,
-        public selectionService: IgxGridSelectionService,
-        public colResizingService: IgxColumnResizingService,
-        @Inject(IGX_GRID_SERVICE_BASE) public gridAPI: GridServiceType,
+        public readonly validation: IgxGridValidationService,
+        /** @hidden @internal */
+        public readonly selectionService: IgxGridSelectionService,
+        protected colResizingService: IgxColumnResizingService,
+        @Inject(IGX_GRID_SERVICE_BASE) public readonly gridAPI: GridServiceType,
         protected transactionFactory: IgxFlatTransactionFactory,
         private elementRef: ElementRef<HTMLElement>,
         protected zone: NgZone,
+        /** @hidden @internal */
         @Inject(DOCUMENT) public document: any,
-        public cdr: ChangeDetectorRef,
+        public readonly cdr: ChangeDetectorRef,
```

Comment:
Ideally, would've loved to make the all the services (e.g. `selectionService`) `protected`, but there's quite a bit of usage internally from other components and tests and not enough time to spend refactoring those to inject the service properly.
There's also a few like `gridAPI` and `navigation` that also have usage in docs & samples, so those will need to be evaluated and go through a deprecation process first, thus can't really hide those for now. Same for `cdr` for example